### PR TITLE
LPS-201571: Avoid NPE when retrieving unrelated Object Entries with related API Prroperties

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Avoid NPE when the object entry is `null` in `EndpointHelper`.
- Add test case for that.

**Which modules are needed to deploy?**
- Deploy `headless-builder-api` and `headless-builder-impl`.

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPS-201571

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java [hunk 1]
```diff
@@ -1140,12 +1140,38 @@ public void testGetWithRelatedModel() throws Exception {
...
-		JSONObject firstItemJSONObject = itemsJSONArray.getJSONObject(0);
+		JSONObject jsonObject = itemsJSONArray.getJSONObject(0);
 
-		JSONArray apiEndpointsERCJSONArray = firstItemJSONObject.getJSONArray(
+		JSONArray apiEndpointsERCJSONArray = jsonObject.getJSONArray(
 			"APIEndpointsERC");
 
 		Assert.assertNotEquals("", apiEndpointsERCJSONArray.getString(0));
+
+		// Unrelated object entry
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java:1150]: This test is passing without deploying your changes. It should fail.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java [hunk 2]
```diff
@@ -1140,12 +1190,38 @@ public void testGetWithRelatedModel() throws Exception {
...
+
+		HTTPTestUtil.invokeToJSONObject(
```
Comment (4lejandrito) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java:1202]: Is this still needed?

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by Hijky** [comment]
The failures are not related to your PR, it's okay to forward it.

**Comment by sergiojimcos** [comment]
ci:forward:force

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java [hunk 3]
```diff
@@ -145,6 +146,10 @@ private Object _getRelatedObjectValue(
 			return objectEntry.getPropertyValue(property.getSourceFieldName());
```
Comment (ccorreagg) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java:146]: I think the if clause validating the null value of the objectEntry should be the first one, as here you are acessing the objectEntry

Comment (sergiojimcos) [reply modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java:146]: Make sense! Closing the Brian's PR and resend

**Comment by sergiojimcos** [comment]
Sent to Brian here: https://github.com/brianchandotcom/liferay-portal/pull/143852

**Pull request closed without merging**